webpack/assets/javascripts/react_app/components/factChart/factChartModal/index.js

```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import { connect } from 'react-redux';
+import { Button, Modal } from 'react-bootstrap';
+import PieChart from '../../common/charts/PieChart/';
+import helpers from '../../../common/helpers';
+import * as FactChartActions from '../../../redux/actions/factCharts/';
+import Loader from '../../common/Loader';
+import MessageBox from '../../common/MessageBox';
+
+class FactChartModal extends React.Component {
+  constructor(props) {
+    super(props);
+    helpers.bindMethods(this, ['closeModal']);
+  }
+
+  closeModal() {
+    this.props.closeModal();
+  }
+
+  render() {
+    const { factChart } = this.props;
+
+    const chart = <PieChart data={factChart.chartData} key={`pie-chart-${factChart.id}`}/>;
+
+    const error = (
+      <MessageBox
+        msg={__('No data available')}
+        icontype="error-circle-o"
+        key={`message-${factChart.id}`}
+      />
+    );
+
+    return (
+      <Modal show={factChart.showModal} onHide={this.closeModal}>
+        <Modal.Header>
+          <Modal.Title>
+            <b>{__(`Fact distribution chart - ${factChart.title}`)}</b>
+          </Modal.Title>
+        </Modal.Header>
+        <Modal.Body>
+          <Loader status={factChart.loaderStatus}>{[chart, error]}</Loader>
+        </Modal.Body>
+        <Modal.Footer>
+          <Button onClick={this.closeModal}>Close</Button>
+        </Modal.Footer>
+      </Modal>
+    );
+  }
+}
+
+const mapStateToProps = (state, ownProps) => ({
+  factChart: ownProps.factChart || state.factChart,
```

Comment:
when do you get ownProps with factChart vs state?

Reply:
this makes the componenet also resuable by just passing props without dealing with state
initially this was sort of a workaround to get storybook to display the component as we couldn't reach redux state in storybook
@sharvit